recipe/meta.yaml

```diff
@@ -138,18 +138,25 @@ outputs:
       requires:
         - pip
       commands:
-        - mamba --help
-        - if exist %PREFIX%\condabin\mamba.bat (exit 0) else (exit 1)  # [win]
-        - test -f ${PREFIX}/etc/profile.d/mamba.sh  # [linux]
-        # these tests work when run on win, but for some reason not during conda build
-        - mamba create -n test_py2 python=2.7 --dry-run  # [unix]
-        - mamba clean --all --dry-run  # [unix]
-        - mamba repoquery whoneeds conda  # [unix]
-        - mamba install xtensor xsimd -c conda-forge --dry-run  # [linux and x86_64]
+        - test -f ${PREFIX}/condabin/mamba  # [unix]
+        - test -f ${PREFIX}/etc/profile.d/mamba.sh  # [unix]
+        - if not exist %PREFIX%\condabin\mamba.bat exit 1  # [win]
+
+        # Ensure we run the executable from the prefix and not from the base installation.
+        - test "$(command -v mamba)" = "${PREFIX}/bin/mamba"  # [unix]
+        - cmd /q /c "for /f "delims=" %%f in ('C:\Windows\System32\where mamba.bat') do if "%%~dpf"=="%LIBRARY_BIN%\" (exit 0) else (exit 1)"  # [win]
+        - cmd /q /c "for /f "delims=" %%f in ('C:\Windows\System32\where mamba.exe') do if "%%~dpf"=="%SCRIPTS%\" (exit 0) else (exit 1)"  # [win]
+
+        # We need to run with "call" on Windows since "mamba" runs the batch file; make "call" a passthrough function on Unixes.
+        - call() { "${@}" ; }  # [unix]
+        - call mamba --help
+        - call mamba create -n test_py37 python=3.7 --dry-run
```

Comment:
`python=2.7` can't be installed on Windows anymore since `vc=9` is on the `cf201901` label.
We'd have to use that label (or `defaults`, I guess) in there to make it work.
But... we could also just not care too much about `python=2.7` anymore and use another (by now old) version.